gcp-implementation-demos/docai-contract-demo/cloud-functions/main.py

```diff
@@ -0,0 +1,165 @@
+import base64
+import re
+import os
+import json
+from datetime import datetime
+from google.cloud import bigquery
+from google.cloud import documentai_v1beta3 as documentai
+from google.cloud import storage
+from google.cloud import pubsub_v1
+
+# Read environment variables
+project_id = os.environ.get("GCP_PROJECT")
+location = os.environ.get("PARSER_LOCATION")
+processor_id = os.environ.get("PROCESSOR_ID")
+timeout = int(os.environ.get("TIMEOUT"))
+dataset_name = os.environ.get("BQ_DATASET_NAME")
+table_name = os.environ.get("BQ_TABLE_NAME")
+
+# Set variables
+address_substring = "address"
+gcs_output_uri = f"gs://{project_id}-output-acme-contracts"
+gcs_output_bucket_name = f"{project_id}-output-acme-contracts"
+gcs_archive_bucket_name = f"{project_id}-archived-acme-contracts"
+gcs_rejected_bucket_name = f"{project_id}-rejected-acme-contracts"
+name = f"projects/{project_id}/locations/{location}/processors/{processor_id}"
+accepted_file_types = [
+    "application/pdf",
+    "image/jpg",
+    "image/png",
+    "image/gif",
+    "image/tiff",
+    "image/jpeg",
+    "image/tif",
+    "image/webp",
+    "image/bmp",
+]
+
+# Create GCP clients
+docai_client = documentai.DocumentProcessorServiceClient()
+storage_client = storage.Client()
+bq_client = bigquery.Client()
+pub_client = pubsub_v1.PublisherClient()
+
+
+def write_to_bq(dataset_name, table_name, entities_extracted_dict):
+    dataset_ref = bq_client.dataset(dataset_name)
+    table_ref = dataset_ref.table(table_name)
+
+    row_to_insert = []
+    row_to_insert.append(entities_extracted_dict)
+    json_data = json.dumps(row_to_insert, sort_keys=False)
+
+    # Convert to a JSON Object
+    json_object = json.loads(json_data)
+
+    job_config = bigquery.LoadJobConfig(
+        source_format=bigquery.SourceFormat.NEWLINE_DELIMITED_JSON,
+        ignore_unknown_values=True,
+        schema=[
+            bigquery.SchemaField("document_name", "STRING"),
+            bigquery.SchemaField("agreement_date", "DATE"),
+            bigquery.SchemaField("arbitration_venue", "STRING"),
+            bigquery.SchemaField("confidentiality_clause", "STRING"),
+            bigquery.SchemaField("effective_date", "DATE"),
+            bigquery.SchemaField("expiration_date", "DATE"),
+            bigquery.SchemaField("governing_law", "STRING"),
```

Comment:
Is there any way to change this to use a schema.json file or something? Seems like this is less reusable